src/hash.java

```diff
@@ -0,0 +1,21 @@
+
+/**
+ * Created by david on 4/19/16.
+ */
+abstract interface hash {
+
```

Comment:
Need to have an overload for the [] operator to support usage like this:
hashmap['my key'] = 'some value'